bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/handler/InsteonDeviceHandler.java

```diff
@@ -136,8 +135,6 @@ public InsteonDeviceHandler(Thing thing) {
     public void initialize() {
         config = getConfigAs(InsteonDeviceConfiguration.class);
 
-        updateStatus(ThingStatus.UNKNOWN);
```

Comment:
Why do you remove that line ?

The initialize method must set a thing status before exiting. I thing this is mentioned in the documentation.
If you remove that line, this will not be the case. For the thing manager, your thing will be in INITIALIZING state ... until the thread you created sets a new state.

Reply:
According to https://www.openhab.org/docs/concepts/things.html#thing-status, INITIALIZING is the correct state:

**INITIALIZING** This status is assigned while the binding initializes the Thing. It depends on the binding how long the initializing process takes. Commands sent to Channels will not be processed.

**UNKNOWN** The handler is fully initialized but due to the nature of the represented device/service it cannot really tell yet whether the Thing is ONLINE or OFFLINE. Therefore the Thing potentially might be working correctly already and may or may not process commands. But the framework is allowed to send commands, because some radio-based devices may go ONLINE if a command is sent to them. The handler should take care to switch the Thing to ONLINE or OFFLINE as soon as possible.

Reply:
Sorry, I forgave, I am probably not enough patient.
@jlaur : please take the lead and merge if that is fine for you.